### PR TITLE
[SPARK-26083][k8s] Add Copy pyspark into corresponding dir cmd in pyspark Dockerfile

### PR DESCRIPTION
**Pull request by AzureQ**
When I try to run `./bin/pyspark` cmd in a pod in Kubernetes(image built without change from pyspark Dockerfile), I'm getting an error:
```
$SPARK_HOME/bin/pyspark --deploy-mode client --master k8s://https://$KUBERNETES_SERVICE_HOST:$KUBERNETES_SERVICE_PORT_HTTPS ...
Python 2.7.15 (default, Aug 22 2018, 13:24:18)
[GCC 6.4.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
Could not open PYTHONSTARTUP
IOError: [Errno 2] No such file or directory: '/opt/spark/python/pyspark/shell.py'
```
This is because `pyspark` folder doesn't exist under `/opt/spark/python/`

## What changes were proposed in this pull request?

Added `COPY python/pyspark ${SPARK_HOME}/python/pyspark` to pyspark Dockerfile to resolve issue above.

## How was this patch tested?
Google Kubernetes Engine

### CODE REVIEW

**Comment by ifilonenko** [comment]
I see this customization to be specific towards how you build your custom Docker image. Unless it is relevant towards testing, we are trying to keep the default Docker image as lightweight as possible (as long as it passes our test cases). Unless one of the commiters sees this as an important thing to include in the default image, I believe it to be a customization.

**Comment by AzureQ** [comment]
> I see this customization to be specific towards how you build your custom Docker image. Unless it is relevant towards testing, we are trying to keep the default Docker image as lightweight as possible (as long as it passes our test cases). Unless one of the committers sees this as an important thing to include in the default image, I believe it to be a customization.

This is to make the default Docker image work properly, not customization at all. Without this change, the docker image of pyspark does not work properly in "client mode" while spark image and rspark image are working properly.

**Comment by ifilonenko** [comment]
> This is fine, but please file a bug.

Okay, as such, @AzureQ could you add an integration test to `ClientModeTestsSuite`

**Comment by AzureQ** [comment]
@ifilonenko I added the test case in `PythonTestsSuites` since it's pyspark specific.

**Comment by vanzin** [comment]
retest this please

File: resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/PythonTestsSuite.scala [hunk 1]
```diff
@@ -89,6 +94,90 @@ private[spark] trait PythonTestsSuite { k8sSuite: KubernetesSuite =>
...
+
+  test("Run PySpark shell", k8sTestTag) {
+    val labels = Map("spark-app-selector" -> driverPodName)
+    val driverPort = 7077
+    val blockManagerPort = 10000
+    val driverService = testBackend
+      .getKubernetesClient
+      .services()
+      .inNamespace(kubernetesTestComponents.namespace)
+      .createNew()
```
Comment (vanzin) [inline resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/PythonTestsSuite.scala]: This doesn't seem to follow the usual indentation used in this module, which makes these calls easier to read. e.g.

https://github.com/apache/spark/blob/master/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/PodTemplateConfigMapStep.scala#L33

Comment (AzureQ) [reply resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/PythonTestsSuite.scala]: Thanks for pointing it out. I'll work on it. Was following https://github.com/apache/spark/blob/master/resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/ClientModeTestsSuite.scala

**Comment by AzureQ** [comment]
@vanzin @ifilonenko I'm able to capture stdout of `spark-shell` but not `pyspark` and `sparkR`. Need more investigation on it. Do you happen to know why?

**Comment by vanzin** [comment]
When I tried to write automated tests for pyspark in the past it was kind of a pain. It doesn't work the way you expect unless you have a pseudo-terminal, apparently.

Maybe try to write a test script around it using `pty.spawn` or `pty.fork`. But to be frank, that sounds more complicated than it's worth...

**Comment by vanzin** [comment]
I'm ok with that.

**Comment by vanzin** [comment]
Merging to master.

**Comment by vanzin** [comment]
Actually I lied.

Could you update the `create_dev_build_context` function in `docker-image-tool.sh` to copy this new directory? You can run the script from your build directory to test it.

You should also revert the change to the python it.

**Comment by AzureQ** [comment]
done the change...not sure why the test build errors out with a hive related issue

**Comment by vanzin** [comment]
retest this please

**Comment by vanzin** [comment]
ok, I give up on flaky tests.

Merging to master.

**Comment by gatorsmile** [comment]
@vanzin Could you create a JIRA for this flaky test?

**Comment by AzureQ** [comment]
> ok, I give up on flaky tests.
> 
> Merging to master.

Thanks!

**Comment by vanzin** [comment]
I thought there was already one for that Hive suite failing... SPARK-23622?

**Pull request closed without merging**